### DISCUSSION TITLE
First party support for PWAs

### DISCUSSION THREAD

**Discussion by clicktodev** [vercel/next.js#42676] [Ideas]
### Describe the feature you'd like to request

I would like to see Nextjs offer a way to configure PWAs without relying on third party packages.

### Describe the solution you'd like

Directly enable and configure PWAs for next apps from the next config file.

### Describe alternatives you've considered

I tried [next-pwa](https://github.com/shadowwalker/next-pwa) but it has some issues and doesn't officially support next 13.

**Comment by angelhodar** [comment]
I would like to up this because I think it could be a really nice feature to include in the framework and not rely on external packages. A recent great example would be next-seo, where now with the Metadata API we have powerful SEO capabilities built-in.

A few months before writing this and with the release of IOS 16.4, Safari now  allows to install PWA and receive web push notifications. I really dont understand why the concept of PWA is not more popular  in the web development ecosystem, it provides a really cheap and maintainable way to create mobile experiences. I think many of the apps out there doesnt really need the performance improvements that native languages and frameworks provide against a PWA, except if they need native APIs for example.

Imagine that with a single codebase in nextjs you can have a web and a mobile-like app. I dont know if the NextJS team have a blog post or some thoughts about this, but the future of PWA seems very promising giving the actual state of web standards adopted across frameworks

**Comment by Zwyx** [comment]
I came here for exactly that.

[`next-pwa`](https://github.com/shadowwalker/next-pwa) — the go-to PWA package for Next — is not maintained. It doesn't support the app directory and its example for i18n is broken. The author seems to be unreachable.

The fork that seems to be the most active is https://github.com/DuCanhGH/next-pwa but it's made by only one person, which gives it some chances of ending up the same way unfortunately.

It would be awesome to have Next.js provide support for PWAs.

**Reply by coopbri** [reply]
Agreed -- I also made an issue about this on the `next-pwa` repo: https://github.com/shadowwalker/next-pwa/issues/482

**Comment by AllanOliveiraM** [comment]
YES, please!! My app relies heavily on PWA to function, and it currently uses next-pwa. However, I'm facing several issues due to the lack of updates, and this could potentially derail my entire project, and I already have many clients.

**Comment by danielbarion** [comment]
PWA example working: 

https://github.com/vercel/next.js/pull/49065/files

`@imbios/next-pwa`

**Comment by feynon** [comment]
Any updates on this?

**Comment by leerob** [comment]
Hey everyone. We now have updated documentation on building PWAs with Next.js.

https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps

**Reply by clicktodev** [reply]
Thanks for the great news! Can’t wait to try it out 🚀

**Reply by xbdirisxk** [reply]
wow, finally, it took years but thank you :)

**Reply by clicktodev** [reply]
@leerob i think there's an issue with doc search indexing because searching for pwa doesn't return anything
![image](https://github.com/user-attachments/assets/399ab3da-5d05-4918-b4a3-b67955e9a179)

**Reply by angelhodar** [reply]
Hi Lee, great article! Just one little question: what would be the value of the ``endpoint`` prop of the PushSubscription object as there is no push service involved like FCM or Apple Push. Are these services not needed?
